avocado/core/result.py

```diff
@@ -133,12 +137,14 @@ def start_tests(self):
         Called once before any tests are executed.
         """
         self.tests_run += 1
+        self._result_events_dispatcher.map_method('pre_tests', self.job)
```

Comment:
My first version just defined the hooks in the remote runner, but when I looked at it I prefer having it inside the Result class. It makes sense because Result is not the place where we store the details and I don't like having 2 lines on each hook (Result.start_test; dispatcher.start_test), therefor I prefer having the hooks inside Result.